### PR TITLE
[BEAM-4798] Fix IndexOutOfBoundsException in Flink runner

### PR DESCRIPTION
**Pull request by aljoscha**
This is an addition on top of #5996 that fixes the test so that it would have caught the bug that @aromanenko-dev fixed in #5996.

CC @aromanenko-dev 

@tweise Maybe this is relevant to your work on portability but I don't think so. Neverthelesse, doesn't hurt to take a look. 😅 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | --- | --- | --- | ---

### CODE REVIEW

**Comment by aljoscha** [comment]
Run Flink ValidatesRunner

**Comment by aljoscha** [comment]
retest this please

File: runners/flink/src/test/java/org/apache/beam/runners/flink/streaming/TestCountingSource.java [hunk 1]
```diff
@@ -109,8 +115,13 @@ public int getShardNumber() {
...
   public List<TestCountingSource> split(int desiredNumSplits, PipelineOptions options) {
     List<TestCountingSource> splits = new ArrayList<>();
-    int numSplits = allowSplitting ? desiredNumSplits : 1;
-    for (int i = 0; i < numSplits; i++) {
+    int actualNumSplits;
+    if (fixedNumSplits == -1) {
+      actualNumSplits = desiredNumSplits;
+    } else {
+      actualNumSplits = fixedNumSplits;
+    }
```
[the remark this answers is not in the document]
Comment (aljoscha) [reply runners/flink/src/test/java/org/apache/beam/runners/flink/streaming/TestCountingSource.java]: yes, fixing

**Comment by aljoscha** [comment]
retest this please

**Comment by aljoscha** [comment]
retest this please

**Comment by aromanenko-dev** [comment]
Thank you @aljoscha for fixing the tests issue, LGTM

**Comment by aljoscha** [comment]
thanks @tweise !

And thanks @aromanenko-dev for discovering and fixing it!

**Pull request merged**